library/src/main/java/com/google/android/exoplayer/audio/AudioTrack.java

```diff
@@ -414,10 +412,15 @@ public int handleBuffer(ByteBuffer buffer, int offset, int size, long presentati
       bytesWritten = writeNonBlockingV21(audioTrack, buffer, temporaryBufferSize);
     }
 
-    temporaryBufferSize -= bytesWritten;
-    submittedBytes += bytesWritten;
-    if (temporaryBufferSize == 0) {
-      result |= RESULT_BUFFER_CONSUMED;
+    if (bytesWritten < 0) {
+        Log.w(TAG, "AudioTrack.write returned error code: " + bytesWritten);
```

Comment:
No need to change as part of this request, since it's not a regression, but I wonder about the logic of just printing a warning here. The AudioTrack documentation suggests failures will be permanent (particularly ERROR_INVALID_OPERATION and ERROR_DEAD_OBJECT), in which case just claiming the buffer was consumed and trying to continue isn't going to help.

Did you actually see any instances where you were able to survive one of these errors successfully? If not, I'll follow this change with another one that changes this code to propagate the error and fail playback instead.